File: assignments/pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // get input from user
    float dollars;
    do
    {
        dollars = get_float("How much do I owe?\n");
    }
    while (dollars < 0);

    // convert to cents
    int cents = round(dollars * 100);

    // variable to store minimum number of coins
    int coins = 0;

    // add coins till needed
    while (cents > 0)
    {
        if (cents >= 25)
        {
            cents -= 25;
            coins++;
        }
        else if (cents >= 10)
        {
            cents -= 10;
            coins++;
        }
        else if (cents >= 5)
        {
            cents -= 5;
            coins++;
        }
        else
        {
            cents--;
            coins++;
        }
    }

    // print result
    printf("%i\n", coins);
}

File: assignments/pset6/cash/cash.py
from cs50 import get_float

while True:
    dollars = get_float("How much do I owe?\n")
    if dollars >= 0:
        break

cents = round(dollars * 100)

coins = 0
while cents > 0:
    if cents >= 25:
        cents -= 25
        coins += 1
    elif cents >= 10:
        cents -= 10
        coins += 1
    elif cents >= 5:
        cents -= 5
        coins += 1
    else:
        cents -= 1
        coins += 1

print(coins)